applications/asset_tracker_v2/src/cloud/lwm2m_integration/lwm2m_integration.c

```diff
@@ -251,6 +251,70 @@ static int modem_mode_request_cb(enum lte_lc_func_mode new_mode, void *user_data
 	return CONFIG_LWM2M_INTEGRATION_MODEM_MODE_REQUEST_RETRY_SECONDS;
 }
 
+static int firmware_update_state_cb(uint8_t update_state)
+{
+	int err;
+	uint8_t update_result;
+	struct cloud_wrap_event cloud_wrap_evt = { 0 };
+
+	/* Get the firmware object update result code */
+	err = lwm2m_engine_get_u8("5/0/5", &update_result);
+	if (err) {
+		LOG_ERR("Failed getting firmware result resource value");
+		cloud_wrap_evt.type = CLOUD_WRAP_EVT_ERROR;
+		cloud_wrap_evt.err = err;
+		cloud_wrapper_notify_event(&cloud_wrap_evt);
+		return 0;
+	}
+
+	switch (update_state) {
+	case STATE_IDLE:
+		LOG_DBG("STATE_IDLE, result: %d", update_result);
+
+		/* If the FOTA state returns to its base state STATE_IDLE, the FOTA failed. */
+		cloud_wrap_evt.type = CLOUD_WRAP_EVT_FOTA_ERROR;
+		break;
+	case STATE_DOWNLOADING:
+		LOG_DBG("STATE_DOWNLOADING, result: %d", update_result);
+		cloud_wrap_evt.type = CLOUD_WRAP_EVT_FOTA_START;
+		break;
+	case STATE_DOWNLOADED:
+		LOG_DBG("STATE_DOWNLOADED, result: %d", update_result);
+		return 0;
+	case STATE_UPDATING:
+		LOG_DBG("STATE_UPDATING, result: %d", update_result);
+		return 0;
+	default:
+		LOG_ERR("Unknown state: %d", update_state);
+		cloud_wrap_evt.type = CLOUD_WRAP_EVT_FOTA_ERROR;
+		break;
+	}
+
+	cloud_wrapper_notify_event(&cloud_wrap_evt);
+	return 0;
+}
+
+static int firmware_update_cb(uint16_t obj_inst_id, uint8_t *args, uint16_t args_len)
+{
+	ARG_UNUSED(obj_inst_id);
+	ARG_UNUSED(args);
+	ARG_UNUSED(args_len);
+
+	int err;
+	struct cloud_wrap_event cloud_wrap_evt = {
+		.type = CLOUD_WRAP_EVT_FOTA_DONE
+	};
+
+	err = lwm2m_firmware_apply_update(0);
```

Comment:
You could also pass the `obj_inst_id` here, it doesn't really matter currently as the multi-instanced FOTA isn't used, but might save some debugging time in future when the multi-instanced FOTA is defined.

Reply:
Sounds good, ill make sure to update the input arg.